packages/repository/src/repositories/relation.factory.ts

```diff
@@ -38,18 +35,18 @@ export interface HasManyDefinition extends RelationDefinitionBase {
  * relation attached to a datasource.
  *
  */
-export function hasManyRepositoryFactory<SourceID, T extends Entity, ID>(
+export function relationRepositoryFactory<SourceID, T extends Entity, ID>(
```

Comment:
Not sure if the rename is better.

Reply:
the rename makes sense to me since in the future we're going to be using the same factory function to produce other constrained repositories of different relations

Reply:
> he rename makes sense to me since in the future we're going to be using the same factory function to produce other constrained repositories of different relations

Let's defer the rename until we actually stat adding support for other relation types please.

I am not sure if it will be possible to have a single factory function, considering that each relation type needs a different repository interface.